ChangeLog

```diff
@@ -77,8 +77,14 @@ Release date: 2021-06-29
 
   Closes #4618
 
+* Improved the Similarity checker performance.
+
+  Close #4120
+
```

Comment:
Can you move this to the `2.10` section?

Reply:
Done. As specified in this section i also added an entry in the `doc/whatsnew/2.10.rst`.